tokio/build.rs

```diff
@@ -19,12 +19,13 @@ const CONST_MUTEX_NEW_PROBE: &str = r#"
 
 const AS_FD_PROBE: &str = r#"
 {
```

Comment:
I think we can use `#[allow(unused_imports)]` (not `#![..]`) on this block.

Reply:
Nope, that doesn't work:

```
djc-2021 fix-as-fd-probe tokio $ git diff
diff --git a/tokio/build.rs b/tokio/build.rs
index 40ee4384..68207afb 100644
--- a/tokio/build.rs
+++ b/tokio/build.rs
@@ -18,14 +18,11 @@ const CONST_MUTEX_NEW_PROBE: &str = r#"
 "#;
 
 const AS_FD_PROBE: &str = r#"
-{
-    #[allow(unused_imports)]
+#[allow(unused_imports)] {
     #[cfg(unix)]
     use std::os::unix::prelude::AsFd as _;
-    #[allow(unused_imports)]
     #[cfg(windows)]
     use std::os::windows::prelude::AsSocket as _;
-    #[allow(unused_imports)]
     #[cfg(target = "wasm32-wasi")]
     use std::os::wasi::prelude::AsFd as _;
 }
```

results in

```
[tokio 1.28.0] error[E0658]: attributes on expressions are experimental
[tokio 1.28.0]  --> <anon>:2:1
[tokio 1.28.0]   |
[tokio 1.28.0] 2 | #[allow(unused_imports)] {
[tokio 1.28.0]   | ^^^^^^^^^^^^^^^^^^^^^^^^
[tokio 1.28.0]   |
[tokio 1.28.0]   = note: see issue #15701 <https://github.com/rust-lang/rust/issues/15701> for more information
[tokio 1.28.0] 
[tokio 1.28.0] error: aborting due to previous error
[tokio 1.28.0] 
[tokio 1.28.0] For more information about this error, try `rustc --explain E0658`.
[tokio 1.28.0] cargo:rustc-cfg=tokio_no_as_fd
```

Reply:
Ah, thanks for checking it. I forgot autocfg uses [the specified expr in rhs of let statement](https://github.com/cuviper/autocfg/blob/e75324cfa5fa0bec4f06f9609ab169bb7747f984/src/lib.rs#L352).